pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -621,6 +870,58 @@ func TestNodeUpdate(t *testing.T) {
 	}
 }
 
+func TestMemberChanged(t *testing.T) {
+	mockController := newMockMulticastController(t, false)
+	_ = mockController.initialize(t)
+	stopCh := make(chan bool)
+	defer close(stopCh)
+	containerA := &interfacestore.ContainerInterfaceConfig{PodNamespace: "nameA", PodName: "podA", ContainerID: "tttt"}
+	containerB := &interfacestore.ContainerInterfaceConfig{PodNamespace: "nameA", PodName: "podB", ContainerID: "mmmm"}
+	for _, tc := range []struct {
+		podUpdateEvent       types.PodUpdate
+		podJoinGroups        map[string][]*interfacestore.ContainerInterfaceConfig
+		groupLeaveEventCount int
+	}{
+		{
+			podUpdateEvent:       types.PodUpdate{PodNamespace: containerA.PodNamespace, PodName: containerA.PodName, IsAdd: true, ContainerID: containerA.ContainerID},
+			groupLeaveEventCount: 0,
+		},
+		{
+			podUpdateEvent: types.PodUpdate{PodNamespace: containerA.PodNamespace, PodName: containerA.PodName, IsAdd: false, ContainerID: containerA.ContainerID},
+			podJoinGroups: map[string][]*interfacestore.ContainerInterfaceConfig{
+				"224.4.5.5": {containerA, containerB},
+				"224.4.5.6": {containerA},
+			},
+			groupLeaveEventCount: 2,
+		},
+	} {
+		mockController.groupEventCh = make(chan *mcastGroupEvent, 100)
+		for group, containers := range tc.podJoinGroups {
+			podTimeMap := make(map[string]time.Time)
+			for _, container := range containers {
+				podTimeMap[agentutil.GenerateContainerInterfaceName(container.PodName, container.PodNamespace, container.ContainerID)] = time.Now().Add(-10 * time.Second)
```

Comment:
Use a variable for Pod interface name, sice it is called more once?

Reply:
Updated. I created a map for storing ContainerInterfaceNames